staging/src/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.proto

```diff
@@ -400,6 +400,9 @@ message PodSandboxStatusRequest {
 message PodSandboxNetworkStatus {
     // IP address of the PodSandbox.
     string ip = 1;
+    // additional ips of the PodSandBox
+    // this list will have all *other* ips
+    repeated string ips = 2;
```

Comment:
Document the first one is the primary IP? ip==ips[0]

Reply:
Thanks. I will, but i don't want to trigger regeneration of proto files now, let keep this one last until we finalize everything else. For now i will keep the comment unresolved.

Reply:
Shouldn't this have metadata too? Otherwise, how will it get passed up to kubelet?

Reply:
Can yo please elaborate? I am not sure i understand your questions.

Reply:
Sure!
So, the kubelet itself isn't really responsible for generating the list of Pod IPs, as you know - that's from the runtime. The runtime actually executes CNI, and returns its understanding of the network status back via the CRI.

One pat of the dual-stack API change was to add IP address metadata *now*, so that it's available later. However, it's likely that the metadata will come from the runtime / CNI, which means it needs to be part of the CRI api proto as well. So here's where it needs to be added.

Make sense?

Reply:
@squeed that got dropped from the initial PR last week for reasons I'll talk about in standup today...

Reply:
May I ask you what the current status of this is? For me it feels pretty strange to fill in two ip/s fields. Is it just to preserve backwards compatibility? If so, I will still break the API here and convert the `string` to a `repeated string`.

Reply:
Can the CRI-plugin really know which is the "primary IP"? IMHO there is no such thing, at least it can't be specified in `host-local` ipam. I think the kubelet(?) should select an address from `Ips` (which may contain more than 2 addresses in the future I guess) of the ClusterIP family (ipv4 or ipv6) and use that as `podIP`.

Reply:
Chatted with @dcbw, and I was more-or-less right. This should still be a `repeated PodIP`, like the core.Pod api above.

Reply:
@uablrek I think you're right. Either the CRI interface itself must specify how the CRI implementation picks the "default pod IP" or kubelet needs to pick the one it wants to use.

Reply:
A couple things:

1) This should be a repeated message, as with the k8s API, so metadata can flow back (even if we don't have the metadata fields yet).

2) This is documented as "additional" IPs.  That is very different from the k8s API, and is maybe OK, but I am honestly not close enough to CRI impl to say for sure.  I think we need someone like @yujuhong to weigh in.  I'll see if I can find her this afternoon so she doesn't have to deal with figuring out all the context.  If it is going to be different semantics, we should make extra effort to comment that here so no future-reader is confused.  If they are going to be the same, we should ALSO make extra effort to comment that here.